rotkehlchen/chain/evm/decoding/aave/common.py

```diff
@@ -123,33 +127,64 @@ def _decode_collateral_events(
             self,
             token: 'EvmToken',
-            tx_log: EvmTxReceiptLog,
-            decoded_events: list['EvmEvent'],
-    ) -> tuple[EvmEvent | None, EvmEvent | None]:
+            context: DecoderContext,
+    ) -> tuple[EvmEvent | None, EvmEvent | None, EvmEvent | None]:
         """Decode aave v2/v3 deposit event. Returns the Deposit and Receive events."""
-        user = hex_or_bytes_to_address(tx_log.data[:32])
-        on_behalf_of = hex_or_bytes_to_address(tx_log.topics[2]) if hex_or_bytes_to_str(tx_log.topics[2]) != '' else None  # noqa: E501
+        user = hex_or_bytes_to_address(context.tx_log.data[:32])
+        on_behalf_of = hex_or_bytes_to_address(context.tx_log.topics[2]) if hex_or_bytes_to_str(context.tx_log.topics[2]) != '' else None  # noqa: E501
         # in the case of needing to wrap the native asset aave uses the
         # (NATIVE_CURRENCY)_GATEWAY contract and the user is the contract address
         if (
             (user not in self.native_gateways and not self.base.is_tracked(user)) and
             (on_behalf_of is None or not self.base.is_tracked(on_behalf_of))
         ):
-            return None, None
+            return None, None, None
         amount = asset_normalized_value(
-            amount=hex_or_bytes_to_int(tx_log.data[32:64]),
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
             asset=token,
         )
+
+        earned_event = None  # find the mint event of aToken
+        for _log in context.all_logs:
+            if (
+                _log.topics[0] == MINT and (on_behalf_of is None or (
+                    on_behalf_of is not None and
+                    hex_or_bytes_to_address(_log.topics[2]) == on_behalf_of
+                )) and
+                user == hex_or_bytes_to_address(_log.topics[1]) and (
+                    balance_increase := asset_normalized_value(
+                        amount=hex_or_bytes_to_int(_log.data[32:64]),
+                        asset=token,
+                    )
+                ) > 0
+            ):  # parse the mint amount and balance_increase
+                context.decoded_events.append(earned_event := self.base.make_event_from_transaction(  # noqa: E501
+                    transaction=context.transaction,
+                    tx_log=_log,
+                    event_type=HistoryEventType.RECEIVE,
+                    event_subtype=HistoryEventSubType.INTEREST,
+                    asset=(a_token := get_or_create_evm_token(
+                        userdb=self.evm_inquirer.database,
+                        evm_address=_log.address,
+                        chain_id=self.evm_inquirer.chain_id,
+                        evm_inquirer=self.evm_inquirer,
+                    )),
+                    balance=Balance(amount=balance_increase),
+                    location_label=user,
+                    notes=f'Receive {balance_increase} {a_token.symbol} as interest earned from {self.label}',  # noqa: E501
```

Comment:
hmm isn't this only for aave v3? I think the v2 logic was different. Just notice it's in the common.py decoder.